File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> map = new HashMap<>();
    for (int i = 0; i < nums.length; i++) {
      int value = target - nums[i];
      if (map.get(value) == null) {
        map.put(nums[i], i);
      } else {
        return new int[]{map.get(value), i};
      }
    }
    return nums;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            current = nums[i]
            complement = target - current
            if complement in map:
                return [i, map[complement]]
            map[current] = i
